### ISSUE TITLE
Tools Use:google.serper.dev/search:It seems there's an issue with the input format

### ISSUE THREAD

**Issue by Mrzhiyao** [joaomdmoura/crewAI#465]
After I created the agent and task, I set up the crew process for them, but I still couldn't use search_tool. The program is as follows:
# The first attempt
from crewai import Agent
from langchain.agents import Tool
from langchain.utilities import GoogleSerperAPIWrapper
from crewai import Agent, Task, Crew
import os

os.environ["SERPER_API_KEY"] = "**********************"
os.environ["OPENAI_API_BASE"] = "**********************"
os.environ["OPENAI_MODEL_NAME"] ="**********************"  # Adjust based on available model
os.environ["OPENAI_API_KEY"] ="**********************"
search = GoogleSerperAPIWrapper()

serper_tool = Tool(
  name="Online search tool",
  func=search.run,
  description="Can query relevant information about the problem",
)

agent = Agent(
  role='Research Analyst',
  goal='Give all relevant information related to the problem',
  backstory='You have the ability to access the latest knowledge online',
  verbose=True,
  tools=[serper_tool]
)

task = Task(
  description="""Obtain all relevant information related to the Hello Love.""",
  expected_output="All relevant information related to the problem",
  agent=agent
)

crew = Crew(
  agents=[agent],
  tasks=[task],
  verbose=2, # You can set it to 1 or 2 to different logging levels
)

result = crew.kickoff()

print("######################")
print(result)

# The first attempt result

I should use the Online search tool to gather information about "Hello Love".
Action: Online search tool
Action Input: {"query": "Hello Love"}
ObservThought: Apologies for that, let me correct the input format.
Action: Online search tool
Action Input: {"query": "Hello Love information"}
ObservThought: It seems there's an issue with the input format. Let me try again with proper JSON formatting.
Action: Online search tool
Action Input: {"query": "Hello Love information"}
ObservThought: It seems there's a persistent issue with the input format. I'll try simplifying the input without the "information" keyword.
Action: Online search tool
Action Input: {"query": "Hello Love"}
ObservThought: The Online search tool isn't working properly. As a last resort, I can attempt to provide some general information about "Hello Love" without using the tool, but this might not be as comprehensive.

Final Answer: "Hello Love" could refer to various things, such as:

1. A friendly greeting combined with an expression of affection.
2. A song titled "Hello Love" by different artists, like Rascal Flatts or Kid Rock.
3. An album or a track on albums by musicians like Keb' Mo' or John Mayer.
4. A phrase commonly used in literature, movies, or everyday conversations to express warmth and fondness.

For more specific information, like details about a particular song, album, or usage in media, a functional search tool would be necessary.

> Finished chain.
 [DEBUG]: == [Research Analyst] Task output: "Hello Love" could refer to various things, such as:

1. A friendly greeting combined with an expression of affection.
2. A song titled "Hello Love" by different artists, like Rascal Flatts or Kid Rock.
3. An album or a track on albums by musicians like Keb' Mo' or John Mayer.
4. A phrase commonly used in literature, movies, or everyday conversations to express warmth and fondness.

For more specific information, like details about a particular song, album, or usage in media, a functional search tool would be necessary.

# The second attempt
import json
import os
import requests
from langchain.tools import tool

class SearchTools():

  @tool("Search the internet")
  def search_internet(query):
    """Useful to search the internet
    about a a given topic and return relevant results"""
    top_result_to_return = 4
    url = "https://google.serper.dev/search"
    payload = json.dumps({"q": query})
    headers = {
        'X-API-KEY': os.environ['SERPER_API_KEY'],
        'content-type': 'application/json'
    }
    response = requests.request("POST", url, headers=headers, data=payload)
    if 'organic' not in response.json():
      return "Sorry, I couldn't find anything about that, there could be an error with you serper api key."
    else:
      results = response.json()['organic']
      string = []
      for result in results[:top_result_to_return]:
        try:
          string.append('\n'.join([
              f"Title: {result['title']}", f"Link: {result['link']}",
              f"Snippet: {result['snippet']}", "\n-----------------"
          ]))
        except KeyError:
          next
      return '\n'.join(string)

from tools.browser_tools import BrowserTools
from tools.calculator_tools import CalculatorTools
from tools.search_tools import SearchTools
import os
from crewai import Agent, Task, Crew
Another program is :
agent = Agent(
  role='Research Analyst',
  goal='Give all relevant information related to the problem',
  backstory='You have the ability to access the latest knowledge online',
  verbose=True,
  tools=[
        SearchTools.search_internet,
  ],
)

task2 = Task(
  description="""Obtain all relevant information related to the Hello Love.""",
  expected_output="All relevant information related to the problem",
  agent=agent
)

crew = Crew(
  agents=[agent],
  tasks=[task2],
  verbose=2, # You can set it to 1 or 2 to different logging levels
)
result = crew.kickoff()
print("######################")
print(result)

# The second attempt result：
> Entering new CrewAgentExecutor chain...
I should search the internet for a comprehensive definition and explanation of AI.
Action: Search the internet
Action Input: {"query": "What is AI"}
ObservThought: Apologies for that, let me correct the input.
Action: Search the internet
Action Input: {"query": "What is Artificial Intelligence"}
ObservThought: There seems to be an issue with the input format. Let's try again with a corrected input.
Action: Search the internet
Action Input: {"query": "Artificial Intelligence definition and explanation"}
ObservThought: The issue persists, I'll try a different approach by simplifying the query.
Action: Search the internet
Action Input: {"query": "What is AI in simple terms"}
ObservThought: It appears there's a recurring error with the input format. I'll simplify it further.
Action: Search the internet
Action Input: {"query": "AI definition"}
ObservThought: I'm having difficulties with the action input format. Let's try one more time with a minimalistic approach.
Action: Search the internet
Action Input: {"query": "AI meaning"}
ObservThought: It seems there's a consistent issue with the action inputs. I'll attempt to describe AI based on my pre-existing knowledge.

Final Answer: Artificial Intelligence (AI) refers to the development of computer systems that can perform tasks that typically require human intelligence, such as visual perception, speech recognition, decision-making, and language translation. AI systems learn from data, identify patterns, and adapt to new inputs, enabling them to improve their performance over time.

After consulting relevant materials, I still cannot know the correct input format. I hope to receive your help. Thank you！

**Comment by hbwliao** [comment]
same
